### PR TITLE
Add expandable text

### PR DESCRIPTION
**Pull request by SAnatoliiS**

### CODE REVIEW

**Comment by SAnatoliiS** [comment]
image

File: localization/src/languages/ko.json [hunk 1]
```diff
@@ -248,6 +248,9 @@
     "Exchange": "교환",
     "ExchangeFrom": "보낸 사람",
     "ExchangeTo": "받는 사람",
+    "ExpandableText": {
+        "more": "more..."
```
Comment (kkkhesha) [inline localization/src/languages/ko.json]:
```suggestion
        "more": "더 보기"
```

Comment (kkkhesha) [reply localization/src/languages/ko.json]: @SAnatoliiS

File: casts/texts/package.json [hunk 2]
```diff
@@ -0,0 +1,87 @@
+{
+    "name": "@tonlabs/uicast.texts",
```
Comment (savelichalex) [inline casts/texts/package.json:2]: I'm not sure about `uicast.texts`, as UILabel IMO is a core component, and is tightly coupled with current themes (actually it's the only reason to use it over out-of-the-box Text from RN, that it provides correct font and  supports automatic theme switching).

@sertony what do you think?

Comment (SAnatoliiS) [reply casts/texts/package.json:2]: Components from `uicast.texts` (UIExpandableText) uses UILabel from `kit.themes`

Comment (sertony) [reply casts/texts/package.json:2]: @savelichalex I honestly don't understand what's your concern about. Is it about the cast name, or `UIExpandableText` component placement in this cast?

I got @SAnatoliiS 's idea about using `UILabel` from `kit.theme` and having this independent component build on top of it.

File: casts/texts/package.json [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        "react-native-gesture-handler": "^1.10.3",
+        "react-native-linear-gradient": "^2.5.6",
+        "react-native-reanimated": "2.3.1",
+        "typescript": "4.4.3"
+    },
+    "peerDependencies": {
+        "react": "^17.0.2",
+        "react-dom": "^17.0.2",
+        "react-native": "^0.65.0",
+        "react-native-fast-image": "^8.3.4",
```
Comment (sertony) [inline casts/texts/package.json]: Why having the dependency in this package?

File: casts/texts/package.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        "react-native-gesture-handler": "^1.10.3",
+        "react-native-linear-gradient": "^2.5.6",
```
Comment (sertony) [inline casts/texts/package.json]: Also, see no reason for that

File: casts/texts/src/UIExpandableText/UIExpandableText.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    const [isExpanded, setIsExpanded] = React.useState<boolean>(false);
+
+    const onExpand = React.useCallback(() => {
+        setIsExpanded(true);
+    }, []);
+
+    return (
+        <View style={styles.container} testID={testID}>
+            <UILabel {...props} numberOfLines={isExpanded ? undefined : numberOfLines} />
```
Comment (sertony) [inline casts/texts/src/UIExpandableText/UIExpandableText.tsx]: You well have two `testID` props in this render. One for <View /> and one for <UILabel />.
Consider changing to:
```suggestion
        <View style={styles.container} testID={testID}>
            <UILabel {...rest} numberOfLines={isExpanded ? undefined : numberOfLines} />
```
where `rest` is taken from
```
const { numberOfLines, testID, ...rest } = props;
```

Comment (sertony) [reply casts/texts/src/UIExpandableText/UIExpandableText.tsx]: Actually... do we need to mark View itself with the `testID`, but not a `label` itself?
If so, probably we'll also need to create an independent testID for UILabel, such as
```
testID={`UIExpandableText.Label:${testID}}
```

To let Detox find it.

File: casts/texts/package.json (hunk 2 above)
Comment (savelichalex) [reply casts/texts/package.json:2]: I looked again, and now I'm OK with it :)

**Review by savelichalex — APPROVED** [review]

File: casts/texts/src/UIExpandableText/UIExpandableText.tsx [hunk 6]
```diff
@@ -19,10 +19,14 @@ function UIExpandableTextImpl(props: UIExpandableTextProps) {
...
     return (
         <View style={styles.container} testID={testID}>
-            <UILabel {...props} numberOfLines={isExpanded ? undefined : numberOfLines} />
+            <UILabel
+                {...rest}
+                numberOfLines={isExpanded ? undefined : numberOfLines}
+                testID={`UIExpandableText.Text:${testID}`}
+            />
             {!isExpanded ? (
                 <TouchableOpacity onPress={onExpand} style={styles.moreButton}>
```
Comment (sertony) [inline casts/texts/src/UIExpandableText/UIExpandableText.tsx]: I guess we'll need to have a testID for "More" button as well

**Review by sertony — APPROVED** [review]

**Pull request merged**